### ISSUE TITLE
Can't reopen in container, maybe because of Docker Resource Saver mode

### ISSUE THREAD

**Issue by bamurtaugh** [microsoft/vscode-remote-release#9211]
- VSCode Version:
Version: 1.85.0-insider (user setup)
Commit: 54b8f6a80aaad7489b75e9293b8d7a1700bed815
Date: 2023-11-17T10:49:40.914Z
Electron: 25.9.4
ElectronBuildId: 25127168
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.25987

- Local OS Version: Windows 11 Insiders
- Remote Extension/Connection Type: Dev Containers (latest pre-release extension, in the latest VS Code Insiders)

- Logs:

[9813 ms] Dev Containers 0.324.0 in VS Code 1.85.0-insider (54b8f6a80aaad7489b75e9293b8d7a1700bed815).
[9812 ms] Start: Run: git rev-parse --abbrev-ref --symbolic-full-name @{u}
[10071 ms] Start: Run: git remote get-url origin
[10310 ms] Start: Run: git remote show origin
[12843 ms] Start: Run: git rev-parse --show-cdup
[13041 ms] Start: Run: git rev-parse --abbrev-ref --symbolic-full-name @{u}
[13243 ms] Start: Run: git cat-file -e origin/bamurtaugh/ignite-container:.devcontainer/devcontainer.json
[13442 ms] Start: Run: C:\Users\brmurtau\AppData\Local\Programs\Microsoft VS Code Insiders\Code - Insiders.exe --ms-enable-electron-run-as-node c:\Users\brmurtau\.vscode-insiders\extensions\ms-vscode-remote.remote-containers-0.324.0\dist\spec-node\devContainersSpecCLI.js read-configuration --workspace-folder c:\Users\brmurtau\OneDrive - Microsoft\Documents\GitHub\devcontainers.github.io --log-level debug --log-format json --config c:\Users\brmurtau\OneDrive - Microsoft\Documents\GitHub\devcontainers.github.io\.devcontainer\devcontainer.json --mount-workspace-git-root
[14151 ms] @devcontainers/cli 0.53.0. Node.js v18.15.0. win32 10.0.25987 x64.
[14151 ms] Start: Run: git rev-parse --show-cdup
[14292 ms] Start: Run: docker ps -q -a --filter label=devcontainer.local_folder=c:\Users\brmurtau\OneDrive - Microsoft\Documents\GitHub\devcontainers.github.io --filter label=devcontainer.config_file=c:\Users\brmurtau\OneDrive - Microsoft\Documents\GitHub\devcontainers.github.io\.devcontainer\devcontainer.json
[21348 ms] Start: Run: docker version --format {{.Server.APIVersion}}

Steps to Reproduce:

1. Open a project with a dev container
2. "Dev Containers: Reopen in Container"
3. :bug: Nothing happens

I tried a bunch of different things (reloading VS Code, closing it completely and reopening it, reverting to an older version of the Dev Containers extension, switching between pre-release vs release versions of the extension, switching between VS Code Insiders and Stable), but nothing worked (or showed me an error message or spinner to indicate anything was happening). 

Interest, I tried connecting to a dev container on a VM via the Dev Containers extension and Remote-SSH, and I was able to connect successfully. This leads me to believe the only variable is Docker on my local machine ended up in "Resource Saver mode" (which claims it should restart/exit the resource saver mode if a container is started).

**Comment by bamurtaugh** [comment]
I believe this resource saver mode issue is confirmed. I was able to restart Docker, and as I restarted it, VS Code showed:

![image](https://github.com/microsoft/vscode-remote-release/assets/25310137/0bf0f848-850a-44f5-936e-0006ab97a926)

This seems to indicate to me the "Reopen in Container" command was stuck, as I had used the command ~5 minutes ago, thought it silently failed, but was surprised to get the above error message once I tried restarting Docker.

**Comment by chrmarti** [comment]
The log suggests that `docker version` was hung. I have tried to repro it with Docker Desktop 4.25.1 for Windows without success so far. Does this still repro on your machine?

**Comment by jdhenry08** [comment]
I'm having the same issue launching a dev container from WSL - everything works fine upon first boot, but if I let docker desktop go into resource saver mode and then try to open a dev container, I get the above popup. Manually exiting resource saver mode and relaunching VSCode fixes the issue.

**Comment by bamurtaugh** [comment]
As an update: I'm having trouble getting Docker back to that same state. I needed to reinstall Docker today (it was stuck on trying to start up), and after reinstalling and restarting my computer, Docker runs but never enters Resource Saver Mode. 

[Docs indicate](https://www.docker.com/blog/docker-desktop-4-22/#:~:text=To%20see%20this%20feature%20in,Resource%20Saver%20mode%20is%20activated.) that there isn't an explicit way to enter this mode, and it should just happen if Docker is "idle for 30 seconds with no containers running" - I left Docker (and even my entire machine) untouched for several minutes with no containers running, but I can't seem to re-enter this mode.

As a final test in the meantime, I tried pausing the Docker engine and using the Dev Containers extension to "Reopen in Container." This worked no problem, and Docker started running again. 

If I'm able to enter Docker Resource Saver mode at some point again (or if folks have tips on how to enter it), I'll be sure to try this scenario again.
